### PR TITLE
Add Ushauri patients to the dimension

### PR DESCRIPTION
**Pull request by Marymary-dev**

### CODE REVIEW

**Comment by DennisGibz** [comment]
@Marymary-dev I have noted the following which is worth changing
1.  On the COALESCE,use UshauriPatientPKHash instead of patientpkhash(remember this is now for patients who are under care who we already have)
2.  Do we need all this joins on the CTE yet what we what are just the patients who are not on care?
  FROM   ods.dbo.Ushauri_Patient AS ushauri
             LEFT JOIN ods.dbo.CT_Patient As patient
              ON ushauri.PatientPKHash=patient.PatientPKHash
              AND ushauri.SiteCode=patient.SiteCode
              LEFT JOIN ods.dbo.ct_patientbaselines AS baselines
              ON ushauri.patientpkhash = baselines.patientpkhash
                AND ushauri.sitecode = baselines.sitecode and baselines.voided=0
             LEFT JOIN ods.dbo.intermediate_artoutcomes AS outcomes
                ON outcomes.patientpkhash = ushauri.patientpkhash
                AND outcomes.sitecode = ushauri.sitecode
                where Patient.PatientPKHash is null and patient.voided=0

**Comment by Marymary-dev** [comment]
> @Marymary-dev I have noted the following which is worth changing
> 
> 1. On the COALESCE,use UshauriPatientPKHash instead of patientpkhash(remember this is now for patients who are under care who we already have)
> 2. Do we need all this joins on the CTE yet what we what are just the patients who are not on care?
>    FROM   ods.dbo.Ushauri_Patient AS ushauri
>    LEFT JOIN ods.dbo.CT_Patient As patient
>    ON ushauri.PatientPKHash=patient.PatientPKHash
>    AND ushauri.SiteCode=patient.SiteCode
>    LEFT JOIN ods.dbo.ct_patientbaselines AS baselines
>    ON ushauri.patientpkhash = baselines.patientpkhash
>    AND ushauri.sitecode = baselines.sitecode and baselines.voided=0
>    LEFT JOIN ods.dbo.intermediate_artoutcomes AS outcomes
>    ON outcomes.patientpkhash = ushauri.patientpkhash
>    AND outcomes.sitecode = ushauri.sitecode
>    where Patient.PatientPKHash is null and patient.voided=0

Thanks @DennisGibz  kindly re-review

**Comment by DennisGibz** [comment]
@Marymary-dev the corrections are now aligned.

One last thing, add the sitetype column on 
1. the merge statement 
- On the first select
- when not merged
- and on the update statement to finish the story.

**Comment by Marymary-dev** [comment]
> @Marymary-dev the corrections are now aligned.
> 
> One last thing, add the sitetype column on
> 
> 1. the merge statement
> 
> * On the first select
> * when not merged
> * and on the update statement to finish the story.

@DennisGibz  kindly take this merge statement up after the merge

**Review by DennisGibz — APPROVED** [review]

**Comment by DennisGibz** [comment]
Approved

File: Scripts/NDWH/ALL DIMENSIONS/load_DimPatient.sql [hunk 1]
```diff
@@ -240,7 +265,7 @@ BEGIN
            FROM   combined_matched_all_programs) AS b
     ON ( a.sitecode = b.sitecode
          AND a.patientpkhash = b.patientpkhash
-		
+		 and a.voided  = b.voided
```
Comment (nobert-mumo) [inline Scripts/NDWH/ALL DIMENSIONS/load_DimPatient.sql:268]: @DennisGibz @Marymary-dev I see we re-introduce the join on voided here. @DennisGibz I had seen in an earlier PR you had removed it. Does it have an impact ?

**Comment by DennisGibz** [comment]
@Marymary-dev  I don't you have pulled the latest changes. You shouldn't be having
 a.sitecode = b.sitecode
         AND a.patientpkhash = b.patientpkhash
		 and a.voided  = b.voided< =====

File: Scripts/NDWH/ALL DIMENSIONS/load_DimPatient.sql (hunk 1 above)
Comment (Marymary-dev) [reply Scripts/NDWH/ALL DIMENSIONS/load_DimPatient.sql:268]: @nobert-mumo  removed the join on voided

Comment (nobert-mumo) [reply Scripts/NDWH/ALL DIMENSIONS/load_DimPatient.sql:268]: @Marymary-dev , not sure, that was supposed to be removed on the join not on the update statement, yeah ?

Comment (Marymary-dev) [reply Scripts/NDWH/ALL DIMENSIONS/load_DimPatient.sql]: sure, removed it on the update

**Review by nobert-mumo — APPROVED** [review]
LGTM.

**Pull request merged**